cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -302,20 +347,155 @@ def _calculate_h3_indexes(
         ["h3_index", "h3_centroid_lat", "h3_centroid_lon", "h3_boundary"]
     ] = spatialized_predictions.apply(_add_h3_index_details, axis=1)
 
-    # Filter out any cells where the projected H3 centroid falls outside of the
+    # Filter out any rows where the projected H3 centroid falls outside of the
     # chunk boundary.
     chunk_boundary = _get_chunk_boundary(study_area_metadata, chunk_metadata)
     spatialized_predictions = spatialized_predictions[
         spatialized_predictions.apply(
-            lambda cell: chunk_boundary.contains(
-                geometry.Point(cell["h3_centroid_lon"], cell["h3_centroid_lat"])
+            lambda row: chunk_boundary.contains(
+                geometry.Point(row["h3_centroid_lon"], row["h3_centroid_lat"])
             ),
             axis=1,
         )
     ]
 
-    # TODO: Add logic for fetching predictions from neighboring chunks for boundary
-    # cells.
+    # Extract rows where the projected H3 cell is not fully contained within the chunk
+    # so we can aggregate prediction values across chunk boundaries.
+    boundary_h3_cells = set(
```

Comment:
should prefer panda's `unique` over a set here since we're in pandas land. Sticking with pandas stuff will all be faster than Python alternatives. 
https://pandas.pydata.org/docs/reference/api/pandas.Series.unique.html#pandas.Series.unique
https://pandas.pydata.org/docs/reference/api/pandas.unique.html